examples/Playground.hs

```diff
@@ -42,6 +45,19 @@ testT = Test
     <*> Toml.table insideT "testX" .= testX
     <*> Toml.maybeT (Toml.table insideT) "testY" .= testY
     <*> Toml.wrapper Toml.text "testN" .= testN
+    <*> eitherT1 .= testE1
+    <*> eitherT2 .= testE2
+  where
+    -- different keys for sum type
+    eitherT1 :: BiToml (Either Integer String)
+    eitherT1 = Toml.match (Toml._Integer >>> Toml._Left)  "either.Left"
+           <|> Toml.match (Toml._String  >>> Toml._Right) "either.Right"
+
+    -- same key for sum type
+    eitherT2 :: BiToml (Either String Double)
+    eitherT2 = ( Toml.match (Toml._String >>> Toml._Left)
+             <!> Toml.match (Toml._Double >>> Toml._Right)
+               ) "either"
```

Comment:
Probably should write about the drawbacks of this case (like when the types of `Left` and `Right` are the same it would take the first one)?

Reply:
Yeah, having same types in constructors of sum types is something that can be improved (but I don't have a good solution at the moment). Current implementation basically tries each constructor one by one. It would be good to have something like _chose constructor based on value_ but I don't know how to support this for now...